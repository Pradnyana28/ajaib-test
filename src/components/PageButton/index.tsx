import React from 'react';
import {
    Button,
} from '@chakra-ui/react';

const PageButton = (props: {
    page: number;
    pageNumber: number;
    handlePaginateButton: (pageNumber: number) => void;
}) => <Button size='sm' isActive={props.page === props.pageNumber} onClick={() => props.handlePaginateButton(props.pageNumber)}>{props.pageNumber}</Button>;

export default PageButton;