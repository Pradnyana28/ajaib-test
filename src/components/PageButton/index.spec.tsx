import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PageButton from '.';

describe('<PageButton />', () => {
    it('renders without crashing', async () => {
        const { container } = render(<PageButton pageNumber={10} />);
        expect(container).toMatchSnapshot();
    });
});