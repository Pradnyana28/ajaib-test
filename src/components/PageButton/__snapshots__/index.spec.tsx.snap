// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageButton /> renders without crashing 1`] = `
<div>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    10
  </button>
</div>
`;
